File: web/src/app/core/services/product.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Product } from '../models/models';

@Injectable({ providedIn: 'root' })
export class ProductService {
  private http = inject(HttpClient);

  getCatalog(locationId: string): Observable<Product[]> {
    return this.http.get<Product[]>(`${environment.apiBaseUrl}/products`, { params: { locationId } });
  }
}

File: mobile/lib/services/product_service.dart
import '../models/product.dart';
import 'api_client.dart';

class ProductService {
  final ApiClient _client = ApiClient();

  Future<List<Product>> getCatalog(String locationId) async {
    final data = await _client.get('/products', query: {'locationId': locationId});
    return (data as List).map((e) => Product.fromJson(e)).toList();
  }
}
